### PR TITLE
[EASY] Make app-data crate not dependent on model crate

### PR DESCRIPTION
**Pull request by Mateo-mro**
# Description
Make app-data crate not dependent on model crate

# Changes
- move `bytesHex` and `appDataHash` to the `app-data` crate
- copy the `orderUid` to the `app-data` crate
- eliminate the dependency of the `model` crate in the `app-data` crate
 
## How to test
1. e2e
2. ci
3. no regression breaking changes

### CODE REVIEW

File: crates/model/src/interaction.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ pub struct InteractionData {
     pub target: H160,
     #[serde_as(as = "HexOrDecimalU256")]
     pub value: U256,
-    #[serde(with = "crate::bytes_hex")]
+    #[serde(with = "app_data::bytes_hex")]
```
Comment (MartinquaXD) [inline crates/model/src/interaction.rs:16]: This should probably still use its own version of bytes_hex`, right?

File: crates/app-data/src/bytes_hex.rs
Comment (MartinquaXD) [inline crates/app-data/src/bytes_hex.rs:1]: Moving that into the `app_data` crate does not seem like the right move. We should probably have some helper crate instead. However, it's unclear to me what are the utilities that should be part of that.

Comment (Mateo-mro) [reply crates/app-data/src/bytes_hex.rs:1]: @MartinquaXD yeah, I agree, I didn't want to make it part of `shared` in order not to depend on it too :thinking: we could have our own crate for `bytes-hex`, in this repo or it is own repo, wdyt?

Comment (fleupold) [reply crates/app-data/src/bytes_hex.rs:1]: For some basic types (e.g. `Eth` etc), we are also fine with code duplication, but given that this involves serialisation, I'd also be in favor of a `hex` crate

Comment (Mateo-mro) [reply crates/app-data/src/bytes_hex.rs:1]: Done!

**Review by squadgazzz — APPROVED** [review]
LGTM

File: crates/bytes-hex/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "bytes-hex"
+version = "0.1.0"
+edition = "2021"
```
Comment (MartinquaXD) [inline crates/bytes-hex/Cargo.toml:4]: This should probably also set the authors and license.

File: crates/bytes-hex/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+serde = { workspace = true }
+serde_json = { workspace = true }
+serde_with = { workspace = true }
+hex = "0.4.3"
```
Comment (MartinquaXD) [inline crates/bytes-hex/Cargo.toml:12]: Is there are reason this doesn't also use the workspace `hex` version?

**Review by MartinquaXD — APPROVED** [review]
Changes look good. Just 2 nits.

File: crates/bytes-hex/Cargo.toml (hunk 3 above)
Comment (Mateo-mro) [reply crates/bytes-hex/Cargo.toml:12]: Thanks, really good point, I didn't realize it was in the workspace! :+1:

File: crates/app-data-hash/LICENSE-APACHE
Comment (MartinquaXD) [inline crates/app-data-hash/LICENSE-APACHE:1]: Instead of copying the licenses into each crate we usually just create links to them.

Comment (Mateo-mro) [reply crates/app-data-hash/LICENSE-APACHE:1]: Oh I see, thanks! I also corrected it in the crates `numbers` which had an invalid symlink!

**Review by MartinquaXD — APPROVED** [review]

**Pull request merged**